mono/metadata/w32process.h

```diff
@@ -53,21 +55,68 @@ typedef struct
 	MonoBoolean error_dialog;
 	gpointer error_dialog_parent_handle;
 	MonoBoolean use_shell_execute;
-	MonoString *username;
-	MonoString *domain;
-	MonoObject *password; /* SecureString in 2.0 profile, dummy in 1.x */
-	MonoString *password_in_clear_text;
-	MonoBoolean load_user_profile;
-	MonoBoolean redirect_standard_input;
-	MonoBoolean redirect_standard_output;
-	MonoBoolean redirect_standard_error;
-	MonoObject *encoding_stdout;
-	MonoObject *encoding_stderr;
+
+	MonoString *unused_username;
+	MonoString *unused_domain;
+	MonoObject *unused_password; /* SecureString in 2.0 profile, dummy in 1.x */
+	MonoString *unused_password_in_clear_text;
+	MonoBoolean unused_load_user_profile;
+	MonoBoolean unused_redirect_standard_input;
+	MonoBoolean unused_redirect_standard_output;
+	MonoBoolean unused_redirect_standard_error;
+	MonoObject *unused_encoding_stdout;
+	MonoObject *unused_encoding_stderr;
+
 	MonoBoolean create_no_window;
-	MonoObject *weak_parent_process;
-	MonoObject *envVars;
+
+	MonoObject *unused_weak_parent_process;
+	MonoObject *unused_envVars;
+
 } MonoW32ProcessStartInfo;
 
+TYPED_HANDLE_DECL (MonoW32ProcessStartInfo);
+
+typedef uint32_t gchandle_t; // FIXME use this more, make it typesafe.
+
+typedef struct _MonoCreateProcessCoop {
+	gunichar2 *filename;
```

Comment:
We could have a struct of ptr/handle/gchandle/length, but elsewhere we treat the  gchandles as an array.